src/__internal__/full-screen-heading/full-screen-heading.component.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import tagComponent from '../../utils/helpers/tags';
+import StyledFullScreenHeading, { StyledHeaderContainer } from './full-screen-heading.style';
+import IconButton from '../../components/icon-button';
+import Icon from '../../components/icon';
+
+const FullScreenHeading = React.forwardRef((props, ref) => {
+  const {
+    children,
+    showCloseIcon,
+    onCancel,
+    ...otherProps
+  } = props;
+
+  const closeIcon = () => {
+    if (!showCloseIcon || !onCancel) return null;
+    return (
+      <IconButton
+        data-element='close'
+        onAction={ onCancel }
+      >
+        <Icon type='close' />
+      </IconButton>
+    );
+  };
+
+  return (
+    <StyledFullScreenHeading
+      { ...otherProps }
+      { ...tagComponent('full-screen-heading', props) }
```

Comment:
Do we need this `tagComponent` function instead of just `data-component`?

Reply:
I didn't write this, just moved it but I think it does in case `data-element` or `data-role` are ever passed as props to `FullScreenHeading`